### PR TITLE
Minor code cleanup

### PR DESCRIPTION
**Pull request by jonahsamost**
- Factored out duplicate code into own function.
- Enabled online-softmax trick to only need to iterate once to find sum and max.
- Removed some unused variables. 

Tested `python -m pufferlib.pufferl train puffer_breakout` to ensure training still stable.

### CODE REVIEW

**Comment by jsuarez5341** [comment]
Manually merged onto new version (files moved) - declined transpose, kept safe_logit. That was exactly what I wanted for sample_logits, thanks!
